src/SD-JWT/Holder.cs

```diff
@@ -10,23 +13,124 @@ public string CreatePresentation(string issuerSignedJwt, Disclosure[] disclosure
+                        throw new InvalidOperationException("Invalid SD-JWT - _sd and ... are reserved claim names");
+
+                    var parent = embeddedSdDigests.Parent?.Parent;
+                    if (parent == null || parent.SelectToken(matchingDisclosure.Name) != null)
+                        throw new InvalidOperationException("Invalid SD-JWT - Disclosure name already exists in the payload");
+                
+                    parent.Add(new JProperty(matchingDisclosure.Name, matchingDisclosure.Value));
+                }
+
+                embeddedSdDigests.Parent?.Remove();
+                ValidateDisclosures(issuerSignedJwtPayload, disclosures, processedDigests);
             }
+
+            var embeddedArrayDigests = issuerSignedJwtPayload.SelectTokens("$..['...']").ToList();
+            if (embeddedArrayDigests.Count > 0)
+            {
+                foreach (var arrayDigests in embeddedArrayDigests)
+                {
+                    if (processedDigests.Any(processedDigest => processedDigest == arrayDigests.ToString()))
+                        throw new InvalidOperationException("Invalid SD-JWT - Digests must be unique");
+                    processedDigests.Add(arrayDigests.ToString());
+                    
+                    var matchingDisclosure = disclosures.Find(disclosure => disclosure.GetDigest() == arrayDigests.ToString());
+
+                    if (matchingDisclosure == null)
+                        arrayDigests.Parent?.Parent?.Remove();
+                    else
+                        arrayDigests.Parent?.Parent?.Replace(matchingDisclosure.Value.ToString());
+                }
             
-            return presentation;
+                ValidateDisclosures(issuerSignedJwtPayload, disclosures, processedDigests);
+            }
+            
+            return issuerSignedJwtPayload;
         }
 
-        public SdJwtDoc ReceiveCredential(string sdJwt)
+        private void IsIssuerSignedJwtValid(string issuerSignedJwt, string jwkJson)
         {
-            /*
-                1. Separate the SD-JWT and the Disclosures in the Combined Format for Issuance.
-                2. Hash all of the Disclosures separately.
-                3. Find the places in the SD-JWT where the digests of the Disclosures are included.
-                   If any of the digests cannot be found in the SD-JWT, the Holder MUST reject the SD-JWT.
-                4. Decode Disclosures and obtain plaintext of the claim values.
-             */
-            return new SdJwtDoc(sdJwt);
+            var jwtHandler = new JwtSecurityTokenHandler();
+            
+            var jwtPayload = Base64UrlEncoder.Decode(issuerSignedJwt.Split('.')[1]);
+            var exp = JObject.Parse(jwtPayload).SelectToken("exp");
+            
+            var validationParameters = new TokenValidationParameters
+            {
+                ValidateIssuer = false,
```

Comment:
Issuer should also be validated: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-05#section-6.1